### DISCUSSION TITLE
How to implement localization in React?

### DISCUSSION THREAD

**Discussion by cronosdevl** [rainbow-me/rainbowkit#1572] [Ideas]
Rainbowkit localization (multiple language option) is available in only Next.js?
I want to use EN, CN, JP languages on my Dapp.
You know, we can't use any hook in index file of React, then it means, not able to implement localization in react?
If you know how to implement, please let me know.
Thanks.

**Comment by DanielSinclair** [comment]
@voldemote The Localization API works in React too! You can use any off-the-shelf package to manage localized routing or to inject a language selection drop-down for users, and then you can pass the user's preference into the `RainbowKitProvider` with the `locale` prop like so:
```tsx
<RainbowKitProvider locale="zh-CN" {...etc}>
    {/* Your App */}
</RainbowKitProvider>
```
Reference the Localization docs [here](https://www.rainbowkit.com/docs/localization) for more information.

**Comment by cronosdevl** [comment]
Hi @DanielSinclair, thank you for your response. 
I have investigated the documentation and found that the RainbowkitProvider is located in the index.jsx (or .tsx) file. However, I encountered an issue when trying to change the language dynamically using hooks like useState, as hooks cannot be used in the index file.

We can set language using locale prop of RainbowkitProvider, but can't change it dynamically into multiple languages like English, Japanese etc.

I implemented dropdown for multiple language option with i18n module, and when it changes, language is changed.
But only rainbowkit wallet modal contents are not changed.  It is my problem.

**Reply by Alexuisis** [reply]
Did you find a solution for this?
